gh-127954: Document `PyObject_DelItemString` (#127986)
python/cpython: Doc/c-api/object.rst

   on failure.  This is equivalent to the Python statement ``del o[key]``.

.. c:function:: int PyObject_DelItemString(PyObject *o, const char *key)

>    This is the same as :c:func:`PyObject_DelItem`, but *key* is

Reviewer: This is the canonical[^1] way to write it (`git grep -i "similar to" Doc/c-api` for inspiration):

```suggestion
   Similar to :c:func:`PyObject_DelItem`, but *key* is
```

[^1]: _"This is the same as"_ is sometimes used, but _"Similar to [...], but"_ is a more common, and also more concise.

>    This is the same as :c:func:`PyObject_DelItem`, but *key* is
>    specified as a :c:expr:`const char*` UTF-8 encoded bytes string,

Reviewer 2: ```suggestion
   This is the same as :c:func:`PyObject_DelItem`, but *key* must be
   a null-terminated :c:expr:`const char*` UTF-8 encoded bytes string,
```

Reviewer 2: I think this is clearer and more precise.

Reviewer 3: Not disagreeing, but this is just copy-paste from the surrounding `PySomething_DoSomethingString` functions. I'd rather we be consistent.

Author: So should I take this advice? Or maybe in the future we can clean up this document, but that will require waiting until this PR is processed.

Reviewer 4: Let's be consistent and keep what you already wrote. If we want to mention NULL terminated `const char *`, then we should instead change everything in the docs in one go, but not in this PR (and I don't think we should do it in the first place)
